### PR TITLE
Improved Generators

### PR DESCRIPTION
**Pull request by PeratX**
### Description

New style for Generators; New populators, objects and biomes

### Reason to modify

It will break the style of old Generator, but it is clearer.

### Tests & Reviews

Tested to generate 2 new maps.

### CODE REVIEW

File: src/pocketmine/level/generator/normal/populator/SugarCane.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			if($successfulClusterCount >= $this->baseAmount){
+				return;
+			}
+		}
+	}
+
+	private function getHighestWorkableBlock($x, $z){
+		for($y = 127; $y >= 0; --$y){
+			$b = $this->level->getBlockIdAt($x, $y, $z);
+			if($b !== Block::DIRT and $b !== Block::GRASS and $b !== Block::SAND and $b !== Block::LEAVES and $b !== Block::LEAVES2){
```
Comment (ishitatsuyuki) [inline src/pocketmine/level/generator/normal/populator/SugarCane.php]: Test your shits before submitting.

**Comment by Leo3418** [comment]
Shall we squash commits? :stuck_out_tongue_winking_eye: :joy:

File: src/pocketmine/level/generator/normal/populator/SugarCane.php (hunk 1 above)
Comment (PeratX) [inline src/pocketmine/level/generator/normal/populator/SugarCane.php]: Tell me where is wrong.

Comment (ishitatsuyuki) [inline src/pocketmine/level/generator/normal/populator/SugarCane.php]: OK, I admire you to work on y=127. Think twice before annoying others.

**Comment by dktapps** [comment]
Sticking by vanilla, NetherrackBiome does not exist. It should be HellBiome.

**Comment by ZedCee** [comment]
If this contains the Swiss-cheese cave gen...
I'm not sure many would like that.

**Comment by PeratX** [comment]
@ZedCee Not contains

**Comment by dktapps** [comment]
Yet to see a single sugarcane in 3 worlds and hundreds of chunks. Hmm.

**Comment by dktapps** [comment]
Mushrooms fail hard, as do cacti and sugarcanes. Seeing things like this, and no sugarcane or cacti.
screenshot_20160828-133456
screenshot_20160828-125959

Rejecting for now.

**Comment by NycuR0** [comment]
@dktapps , @PeratX , hello ( and sorry for multi account @dktapps  , but i can help ) . I can create pull request for fix more in generator ? :) and @dktapps ,is growing blocks because he used id for mushroom_block. Need use for mushroom normaly. You can edit from populator mushroom.

I know @dktapps hate me and my project,but i know how fix more and...yes,i can help.

File: src/pocketmine/tile/MobSpawner.php [hunk 2]
```diff
@@ -152,7 +152,7 @@ public function onUpdate(){
 			if($this->getDelay() <= 0){
 				$success = 0;
 				for($i = 0; $i < $this->getSpawnCount(); $i++){
-					$pos = $this->add(mt_rand() / mt_getrandmax() * $this->getSpawnRange(), mt_rand(-1, 1), mt_rand() / mt_getrandmax() * $this->getSpawnRange());
+					$pos = $this->add(mt_rand() / mt_getrandmax() * $this->getSpawnRange(), 1, mt_rand() / mt_getrandmax() * $this->getSpawnRange());
```
Comment (ishitatsuyuki) [inline src/pocketmine/tile/MobSpawner.php]: Hey, don't touch my spawner code. You are DEFINITELY wrong.

**Comment by Muqsit** [comment]
deleted..

**Comment by dktapps** [comment]
What...

**Comment by TiesGamesNLDev** [comment]
wtf mcregios keeps giving errors

**Comment by dktapps** [comment]
@tiesgamesnldev stop spamming.

**Comment by TiesGamesNL** [comment]
im not spamming

**Pull request closed without merging**